include/s3select.h

```diff
@@ -552,11 +221,24 @@ struct s3select : public bsc::grammar<s3select>
 
   bool aggr_flow;
 
-#define BOOST_BIND_ACTION( push_name ) boost::bind( &push_name::operator(), g_ ## push_name , _1 ,_2)
-#define ATTACH_ACTION_Q( push_name ) {(g_ ## push_name).set_action_q(&m_actionQ); (g_ ## push_name).set_s3select_functions(&m_s3select_functions); (g_ ## push_name).set(&m_s3select_allocator);}
+#define BOOST_BIND_ACTION( push_name ) boost::bind( &push_name::operator(), g_ ## push_name, const_cast<s3select*>(&self), _1, _2)
```

Comment:
why is const_cast necessary here?

Reply:
the push_name::operator() are for building the AST,
the /self/ is const reference of the outer class /s3select/ .
outer(s3select) class contain structures which are updated by methods calls from the inner class (grammar definition)

must remove const from self, before updating it ( self-> ..... = )

Reply:
ok thanks, i see that the `definition` constructor has to take a const reference. you can generally avoid const_cast (which has a risk of undefined behavior if the original object was declared as const) by fixing the constness elsewhere, but it looks like we're stuck with it here

Reply:
exactly, had tried, its "locked" by constructor definition.